iocore/dns/DNS.cc

```diff
@@ -437,9 +437,9 @@ DNSEntry::init(const char *x, int len, int qtype_arg, Continuation *acont, DNSPr
   } else { // T_PTR
     IpAddr const *ip = reinterpret_cast<IpAddr const *>(x);
     if (ip->isIp6()) {
-      make_ipv6_ptr(&ip->_addr._ip6, qname);
+      qname_len = make_ipv6_ptr(&ip->_addr._ip6, qname);
     } else if (ip->isIp4()) {
-      make_ipv4_ptr(ip->_addr._ip4, qname);
+      qname_len = make_ipv4_ptr(ip->_addr._ip4, qname);
```

Comment:
Should you update orig_qname_len as well? It's unclear to me.

Reply:
Probably - the `memcmp` in `get_entry` doesn't use it, which is where the problem is, but might as well be consistent.